source/dataman/CLS/CLSDbUpgrade1Pt1.cpp

```diff
@@ -62,29 +66,29 @@ bool CLSDbUpgrade1Pt1::upgradeImplementation()
 		for (int rowId = tableFacilityNames.count(); rowId > 0; rowId--) {
 			dbResult = databaseToUpgrade_->deleteRow(rowId, "AMFacility_table");
 			if ( dbResult == 0 ){
-				AMErrorMon::alert(this, CLSDBUPGRADE1PT1_COULD_NOT_DELETE_FACILITY, QString("Could not delete the facility (%1) in AMFacility_table").arg(rowId));
+				AMErrorMon::alert(this, CLSDBUPGRADE1PT1_COULD_NOT_DELETE_FACILITY, QString("Could not delete the facility (%1) in %2").arg(rowId).arg("AMFacility_table"));
 				databaseToUpgrade_->rollbackTransaction();
 				return false;
 			}
 		}
 
-		// add the new facility
-		QStringList columns = QStringList() << "AMDbObjectType" << "thumbnailCount" << "thumbnailFirstid" << "name" << "description" << "iconFileName";
-		QVariantList values = QVariantList() << "AMFacility" << 1 << 1 << targetFacilityName_ <<  QString("CLS %1 Beamline").arg(targetFacilityName_) << ":/clsIcon.png";
-		dbResult = databaseToUpgrade_->insertOrUpdate(0, "AMFacility_table", columns, values);
-		if (dbResult == 0) {
-			AMErrorMon::alert(this, CLSDBUPGRADE1PT1_COULD_NOT_INSERT_NEW_FACILITY, QString("Could not insert the facility (%1) in AMFacility_table").arg(targetFacilityName_));
+		// add the new facility and create the thumbnail
+		AMFacility *facility = new AMFacility(targetFacilityName_, QString("CLS %1 Beamline").arg(targetFacilityName_), ":/clsIcon.png");
+		if (!facility->storeToDb(databaseToUpgrade_)) { // this will create Thumbnail as well
```

Comment:
Have you tested this?  In general, you can't use AMDbObject inside of AMDbUpgrades because the upgrade has to be done before all the mappings and relations are established.  You should always need to do the inserts and updates by hand (unfortunately).